src/vite/utils/getGitTimestamp.ts

```diff
@@ -0,0 +1,27 @@
+import { spawn } from 'cross-spawn'
+import fs from 'fs-extra'
+import { basename, dirname } from 'path'
+
+const cache = new Map<string, number>()
+
+export function getGitTimestamp(file: string) {
+  const cached = cache.get(file)
+  if (cached) return cached
+
+  return new Promise<number>((resolve, reject) => {
+    const cwd = dirname(file)
+    if (!fs.existsSync(cwd)) return resolve(0)
+    const fileName = basename(file)
+    const child = spawn('git', ['log', '-1', '--pretty="%ai"', fileName], {
```

Comment:
could you use `statSync` or nah?

Reply:
unfortunately not. `statSync` change time and the most recent commit for the specified file could be different.

Reply:
🫡